src/Sparrow/PortableExceptions.cs

```diff
@@ -0,0 +1,806 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+#if NET6_0_OR_GREATER
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+#endif
+
+namespace Sparrow
+{
+    internal class PortableExceptions
+    {
+
+#if NET6_0_OR_GREATER         
+        // In order to defer the allocation of interpolations we need to do it through a InterpolatedStringHandler.
+        // More details in the following issue: https://github.com/dotnet/runtime/issues/101996
+        // We could remove all the API when this is solved at the JIT level. 
+
+        [InterpolatedStringHandler]
+        public readonly ref struct ConditionalThrowInterpolatedStringHandler
+        {
+            private readonly StringBuilder? _builder;
```

Comment:
Can we use `#nullable enable` in this file? So we won't need to enable it globally but use a file-by-file approach

Reply:
OK, but at least for `Voron`, `Sparrow` and `Corax` expect to see in the next few month added usage of those annotations.